### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by mnath30**
Added the following features:
1. Backend and API calls for Address Management
2. Add, edit and update address functionality
3. User profile page

### CODE REVIEW

File: src/Components/AddressCard/address-card.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.address_name {
+  padding: 0.8rem 0;
+}
+
+.address_btn_section {
+  width: 100%;
+  border-top: 1px solid var(--border-color);
+  /* text-align: center; */
```
Comment (ashishkatiyar2404) [inline src/Components/AddressCard/address-card.css]: remove comments

File: src/Components/AddressCard/address-card.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  margin-top: 0.8rem;
+  /* background-color: inherit; */
```
Comment (ashishkatiyar2404) [inline src/Components/AddressCard/address-card.css]: here also

File: src/Components/AddressCard/address-card.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.address_btn {
+  width: 46%;
+  font-size: 0.8rem;
+  border: none;
+  /* padding: 1.8rem; */
```
Comment (ashishkatiyar2404) [inline src/Components/AddressCard/address-card.css]: from here also

File: src/Components/Modal/AddressModal.jsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          error: "Pincode should be 6 digit number",
+        });
+      }
+    } else {
+      setAddressDetails({
+        ...addressDetails,
+        error: "Fill all details",
+      });
+    }
+  };
```
Comment (ashishkatiyar2404) [inline src/Components/Modal/AddressModal.jsx:67]: you can use switch statement instead of if else

File: src/Components/Modal/modal.css [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  border-radius: 5px;
+}
+
+.modal-btn {
+  margin: 0.4rem;
+  border: 2px solid var(--secondary-color);
+  background-color: inherit;
+}
+
+@media only screen and (max-width: 770px) {
```
Comment (ashishkatiyar2404) [inline src/Components/Modal/modal.css:39]: appreciation: - good to see you have taken care of responsive also.

**Review by ashishkatiyar2404 — APPROVED** [review]
functionality and code look good. You can merge.

**Review by anjali1102 — APPROVED** [review]

**Pull request merged**